src/test/java/co/rsk/bitcoinj/script/FlyoverRedeemScriptParserTest.java

```diff
@@ -273,6 +290,36 @@ public void extractStandardRedeemScriptChunks_whenIsP2shErpRedeemScript_shouldRe
         assertEquals(expectedStandardRedeemScriptChunks, actualRedeemScriptChunks);
     }
 
+    @Test
+    public void hasErpFormat_whenFlyoverStandardRedeemScript_shouldReturnFalse() {
+        assertHasNotErpFormat(flyoverStandardRedeemScript);
+    }
+
+    @Test
+    public void hasErpFormat_whenFlyoverNonStandardErpRedeemScriptHardcoded_shouldReturnFalse() {
+        assertHasNotErpFormat(flyoverNonStandardErpRedeemScriptParserHardcoded);
+    }
+
+    @Test
+    public void hasErpFormat_whenFlyoverNonStandardErpRedeemScript_shouldReturnTrue() {
+        assertHasErpFormat(flyoverNonStandardErpRedeemScript);
+    }
+
+    @Test
+    public void hasErpFormat_whenFlyoverP2shRedeemScript_shouldReturnTrue() {
+        assertHasErpFormat(flyoverP2shErpRedeemScript);
+    }
+
+    private void assertHasErpFormat(Script flyoverRedeemScript) {
+        FlyoverRedeemScriptParser flyoverRedeemScriptParser = (FlyoverRedeemScriptParser) RedeemScriptParserFactory.get(flyoverRedeemScript.getChunks());
```

Comment:
This is not parsing. This is casting a class that you expect to be from a specific type. If the script is a different type you will throw a `ClassCastException`.

I added the casting to make sure the resulted object is a `FlyoverRedeemScriptParser`. But I will change to use the `FlyoverRedeemScriptParser` constructor instead since that's the class we testing here.

Reply:
Updated